src/gui/windows/multiplayer.zig

```diff
@@ -56,17 +56,28 @@ fn join(_: usize) void {
 		ipAddress = "";
 	}
 	if(connection) |_connection| {
+		if (ipAddressEntry.currentString.items.len == 0) {
+			std.log.err("IP address cannot be empty", .{});
+			main.gui.windowlist.notification.raiseNotification("IP address cannot be empty");
+			return;
+		}
 		_connection.world = &main.game.testWorld;
 		main.globalAllocator.free(settings.lastUsedIPAddress);
 		settings.lastUsedIPAddress = main.globalAllocator.dupe(u8, ipAddressEntry.currentString.items);
 		settings.save();
 		main.game.world = &main.game.testWorld;
+		std.log.info("Connecting to server: {s}", .{ipAddressEntry.currentString.items});
 		main.game.testWorld.init(ipAddressEntry.currentString.items, _connection) catch |err| {
-			std.log.err("Encountered error while opening world: {s}", .{@errorName(err)});
+			const formattedError = std.fmt.allocPrint(main.stackAllocator.allocator, "Encountered error while opening world: {s}", .{@errorName(err)}) catch unreachable;
+			std.log.err("{s}", .{formattedError});
+			defer main.stackAllocator.free(formattedError);
```

Comment:
directly after the allocation, so before the print, please.

Reply:
✅